homeassistant/components/alarmdecoder/binary_sensor.py

```diff
@@ -122,14 +122,18 @@ def device_state_attributes(self):
         """Return the state attributes."""
         attr = {CONF_ZONE_NUMBER: self._zone_number}
         if self._rfid and self._rfstate is not None:
-            attr[ATTR_RF_BIT0] = bool(self._rfstate & 0x01)
-            attr[ATTR_RF_LOW_BAT] = bool(self._rfstate & 0x02)
-            attr[ATTR_RF_SUPERVISED] = bool(self._rfstate & 0x04)
-            attr[ATTR_RF_BIT3] = bool(self._rfstate & 0x08)
-            attr[ATTR_RF_LOOP3] = bool(self._rfstate & 0x10)
-            attr[ATTR_RF_LOOP2] = bool(self._rfstate & 0x20)
-            attr[ATTR_RF_LOOP4] = bool(self._rfstate & 0x40)
-            attr[ATTR_RF_LOOP1] = bool(self._rfstate & 0x80)
+            attr.update(
+                {
```

Comment:
It's better to set the existing dict than creating a new dict and then just update the existing dict, throwing away the new dict.

Reply:
Maybe speed is more important? :+1:

Reply:
How would your suggestion look like in code?

Reply:
> Maybe speed is more important? +1

That was my intention, but if it is bad practice or makes the code harder to understand I can revert my changes of course.